### ISSUE TITLE
SQL Server 2022 and Azure SQL support?

### ISSUE THREAD

**Issue by dmoonfire** [fluentmigrator/fluentmigrator#1944]
**Describe the question**

My database team is pushing back on using this library because https://fluentmigrator.github.io/index.html doesn't show SQL Server 2022 and AzureSQL as supported platforms. They are using it as "proof" that this is an unsuitable library to use. The fact we're coding it against 2022 isn't enough for them, only official documentation.

I saw that the documentation says "SQL Server 2016 tests run..." on the bottom. Is there a chance that these are also run against the SQL Server 2022 and AzureSQL platforms?

**Documentation Pages You've Read So Far**

https://fluentmigrator.github.io/index.html (the "Supported databases") tasks.

**Expected benefit**

It would give a clearer signal that this is a viable product for later versions of supported servers.

**Information (please complete the following information):**
 - OS: Windows
 - Platform: .NET Framework 4.8.1, .NET Core [.NET Framework 4.x / .NET Core 2.x , etc]
 - FluentMigrator version: 6.2.0
 - FluentMigrator runner: in-process runner
 - Database Management System: SQL Server
 - Database Management System Version: SQL Server 2022, AzureSQL

**Additional context**
N/A

**Comment by PhenX** [comment]
Hello, I think I'm not wrong if I say that FluentMigrator is by default always compatible with more recent versions of the supported databases (unless proved wrong).

Furthermore all integration tests in this [pull request](https://github.com/fluentmigrator/fluentmigrator/pull/1931) are ran on a SQL Server 2022 image and they all pass.

I cannot tell for Azure SQL though, but as it seems to a be SQL server under the hood, it should work, but you need to try it.

**Comment by dmoonfire** [comment]
I'm coding against SQL Server 2022 and it works for me.

But as I said, the fact *I'm* okay doesn't mean the other teams believe it is a supported platform. Documentation is a big thing for the database teams and that is causing doubt in my decision to use this library. Since the website and documentation doesn't say SQL Server 2022, as far as the database team is concerned, Fluent Migrator doesn't work on SQL Server 2022.

**Comment by fubar-coder** [comment]
@dmoonfire The database support (in FM lingua "processor") is meant as a "minimal feature level" the database is required to support. IIRC, **nothing** in FM itself uses features that are in SQL Server 2016, but not in Azure SQL. IOW: FM does indeed work on SQL Server 2016 and every SQL Server version above 2016 if you use the "sqlserver2016" database processor (target).

You can quote me as the previous maintainer of this project.

**Comment by jzabroski** [comment]
Correct, it is supported.  There may be some bleeding edge features in 2022 that are not directly handled as advanced fluent syntax, but the core functionality will work as expected.  We are currently working on a PR to support token based authentication, which would be one thing many Azure SQL customers may want.
